src/test/java/spoon/test/variable/VariableTest.java

```diff
@@ -191,4 +200,27 @@ public void testVarInLambda() {
         assertEquals("java.lang.Long", lambda.getParameters().get(1).getType().getQualifiedName());
         assertEquals("(var x,var y) -> x + y", lambda.toString()); // we should print var, if it was in the original code
     }
+
+    @ModelTest(value = "./src/test/resources/spoon/test/unnamed/UnnamedVar.java", complianceLevel = 22)
+    void testUnnamedVariable(Factory factory) throws IOException {
+        CtType<?> type = factory.Type().get("spoon.test.unnamed.UnnamedVar");
+        assertThat(type.getMethods()).isNotEmpty();
+        List<String> lines = java.nio.file.Files.readAllLines(type.getPosition().getFile().toPath());
+        for (CtMethod<?> method : type.getMethods()) {
+            List<CtVariable<?>> locals = method.getBody().getElements(new TypeFilter<>(CtVariable.class));
+            assertThat(locals).describedAs(method.getSimpleName()).hasSize(1);
+            CtVariable<?> variable = locals.get(0);
+            assertThat(variable).getSimpleName().isEqualTo("_");
+			if (variable instanceof CtLocalVariable<?> v) {
+				assertTrue(v.isUnnamed());
+			} else if (variable instanceof CtParameter<?> v) {
+				assertTrue(v.isUnnamed());
+			} else if (variable instanceof CtCatchVariable<?> v) {
+				assertTrue(v.isUnnamed());
+			}
+            assertThat(variable).getPosition().isNotEqualTo(SourcePosition.NOPOSITION);
+            String line = lines.get(variable.getPosition().getLine() - 1);
+            assertThat(line).contains(variable.toString());
+        }
+    }
```

Comment:
Something in the indentation here is messed up and mixes spaces and tabs.

Reply:
Seems like this file is cursed.

Reply:
Apparently it still uses spaces…